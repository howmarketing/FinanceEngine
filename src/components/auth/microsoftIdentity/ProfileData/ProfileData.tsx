import React from 'react'

/**
 * Renders information about the user obtained from MS Graph
 * @param props
 */

export const ProfileData: React.FC<{
	[x: string]: any;
	graphData: {
		[x: string]: any;
		mail?: string;
		surname: String;
		id: string | number;
		givenName: String;
		displayName?: string;
		userPrincipalName: String;
	}
}> = (props) => {
	return (
		<div id="profile-div">
			<p><strong>Id: </strong> {props?.graphData?.id || 'No id given'}</p>
			<p><strong>First Name: </strong> {props?.graphData?.givenName || 'No given name'}</p>
			<p><strong>Last Name: </strong> {props?.graphData?.surname || 'No surname given'}</p>
			<p><strong>Principal Name: </strong> {props?.graphData?.userPrincipalName || 'No principal name given'}</p>
			<p><strong>Email: </strong> {props?.graphData?.mail || 'No e-mail name given'}</p>
		</div>
	)
}
