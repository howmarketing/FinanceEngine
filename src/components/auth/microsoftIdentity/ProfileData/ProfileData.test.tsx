import React from "react";
import TestRenderer from 'react-test-renderer';
import { ProfileData } from "components/auth/microsoftIdentity/ProfileData/ProfileData";

it('SignInButton component render', () => {
    const tree = TestRenderer.create(<ProfileData key={'jest-test'} graphData={{ id: "1", givenName: "Test", surname: "Jest. ", userPrincipalName: "Jest test" }} />).toJSON()
    expect(tree).toMatchSnapshot()
})