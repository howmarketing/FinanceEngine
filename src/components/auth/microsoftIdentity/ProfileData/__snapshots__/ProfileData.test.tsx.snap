// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInButton component render 1`] = `
<div
  id="profile-div"
>
  <p>
    <strong>
      Id: 
    </strong>
     
    1
  </p>
  <p>
    <strong>
      First Name: 
    </strong>
     
    Test
  </p>
  <p>
    <strong>
      Last Name: 
    </strong>
     
    Jest. 
  </p>
  <p>
    <strong>
      Principal Name: 
    </strong>
     
    Jest test
  </p>
  <p>
    <strong>
      Email: 
    </strong>
     
    No e-mail name given
  </p>
</div>
`;
